lib/sourceParams.js

```diff
@@ -7,14 +7,28 @@ const fileExists = require('./fileExists')
 const checkParams = require('./checkParams')
 const appModulePath = require('app-module-path')
 const defaultConfig = require('./defaultConfig')
+const logger = require('./logger')()
 let appDir = require('./getAppDir')
-let pkg = require(path.join(appDir, 'package.json'))
-
-pkg.rooseveltConfig = pkg.rooseveltConfig || {}
+let pkg
 
 module.exports = function (app) {
   let params = app.get('params')
 
+  // determine if app has a package.json
+  try {
+    pkg = require(path.join(appDir, 'package.json'))
+  } catch (e) {
+    pkg = {}
+
+    // default folder structure generation to false when it's not set in the constructor
+    if (!params.generateFolderStructure) {
+      logger.warn(`Roosevelt initialized in ${appDir} via constructor with no package.json found. No files will be generated unless param generateFolderStructure is set to true. Only do this if you are sure you want files generated in ${appDir}!`.bold.yellow)
+      params.generateFolderStructure = false
```

Comment:
make sure this is set to true if the app does have a package.json

also make sure to document this param in the README